addons/hr_expense/models/hr_expense.py

```diff
@@ -197,6 +201,59 @@ def _compute_employee_id(self):
             for expense in self:
                 expense.employee_id = self.env.user.with_company(expense.company_id).employee_id
 
+    @api.depends('employee_id', 'product_id', 'total_amount')
+    def _compute_duplicate_expense_ids(self):
+        self.duplicate_expense_ids = False
+
+        expenses = self.filtered(lambda e: e.employee_id and e.product_id and e.total_amount)
+        if expenses.ids:
+            duplicates_query = """
+                WITH expenses AS (
+                    SELECT employee_id, product_id, date, total_amount, company_id, currency_id
+                      FROM hr_expense
+                     WHERE id IN %(expense_ids)s
+                     GROUP BY employee_id, product_id, date, total_amount, company_id, currency_id)
+              SELECT ARRAY_AGG(he.id)
+                FROM hr_expense AS he
+               INNER JOIN expenses AS ex
+                  ON he.employee_id = ex.employee_id
+                 AND he.product_id = ex.product_id
+                 AND he.date = ex.date
+                 AND he.total_amount = ex.total_amount
+                 AND he.company_id = ex.company_id
+                 AND he.currency_id = ex.currency_id
+               GROUP BY he.employee_id, he.product_id, he.date, he.total_amount, he.company_id, he.currency_id
+              HAVING COUNT(id) > 1
```

Comment:
I think this query is doable without CTE.
```suggestion
  SELECT ARRAY_AGG(DISTINCT he.id)
    FROM hr_expense AS he
    JOIN hr_expense AS ex ON he.employee_id = ex.employee_id
                          AND he.product_id = ex.product_id
                          AND he.date = ex.date
                          AND he.total_amount = ex.total_amount
                          AND he.company_id = ex.company_id
                          AND he.currency_id = ex.currency_id
   WHERE ex.id IN %(expense_ids)s
GROUP BY he.employee_id, he.product_id, he.date, he.total_amount, he.company_id, he.currency_id
  HAVING COUNT(he.id) > 1
```
should produce the same output.

Reply:
Indeed it works! Thanks for the suggestion :-)